lib/Cpdf.php

```diff
@@ -2542,7 +2542,7 @@ private function openFont($font)
                             $width = floatval($dtmp['WX']);
 
                             if ($c >= 0) {
-                                if ($c != hexdec($n)) {
+                                if (ctype_xdigit($c) && ctype_xdigit($n)) {
```

Comment:
I don't seem to have the necessary permission to add myself as a reviewer, so until that's resolved I'll add my comments here.

After discussing with Matt, I agree with his sentiment that the above change doesn't truly resolve the issue at hand. What is happening is the hexdec() function expects a hexadecimal string to be passed where instead it's potentially being set as "space".

The proposed change is only confirming that variables $c and $n are hexadecimal digits.

I'd recommend the following solution so that invalid inputs are accounted for e.g. "space" but we are still using the original comparison where possible
`
ctype_xdigit($c) && ctype_xdigit($n) && $c != hexdec($n)
`

Reply:
Here's an example of what I'm referring to in a PHP sandbox: https://onlinephp.io?s=VY2xCsIwFEVnC_2HaylNOzgIDoIWB8HRX5Dw8rQP2jQ0qVTEfzcVBx3P4XLu_uAalyZpkhNqqLU-bWirdpHtzJl3mjibRZpArigpPBxfJiM3CWVOFYoC_85-XOwtazQ8GabZVXjGwsIIl-rYWyNBegvxCMPIqooHeIFbzz-7c29Xdux4EMJdtyPDae_ZfOdv&v=7.4.33